src/ai/backend/test/testcases/session/vfolder_mount.py

```diff
@@ -0,0 +1,109 @@
+import os
+import re
+import shutil
+from dataclasses import dataclass
+from typing import override
+
+from ai.backend.test.contexts.client_session import ClientSessionContext
+from ai.backend.test.contexts.image import ImageContext
+from ai.backend.test.contexts.tester import TestSpecMetaContext
+from ai.backend.test.contexts.vfolder import CreatedVFolderMetaContext
+from ai.backend.test.templates.template import TestCode
+
+
+@dataclass
+class _UploadFileMeta:
+    file_name: str
+    full_path: str
+    base_dir: str
+    content: str
+
+
+class VFolderMountSuccess(TestCode):
+    @override
+    async def test(self) -> None:
+        test_spec = TestSpecMetaContext.current()
+        test_id = test_spec.test_id
+        session = ClientSessionContext.current()
+        image = ImageContext.current()
+        test_name = f"test-{test_id}"
+        vfolder_meta = CreatedVFolderMetaContext.current()
+
+        try:
+            await session.ComputeSession.get_or_create(
+                image=image.name, name=test_name, mounts=[vfolder_meta.name]
+            )
+            file_upload_meta = await self._create_dummy_file()
+
+            await session.VFolder(name=vfolder_meta.name).upload(
+                sources=[file_upload_meta.file_name],
+                basedir=file_upload_meta.base_dir,
+            )
+
+            exec_result = await session.ComputeSession(name=test_name).list_files(
+                path=vfolder_meta.name
+            )
+            self._verify_file_exists(exec_result, file_upload_meta.file_name)
+
+            await session.ComputeSession(name=test_name).download(
+                files=[f"./{vfolder_meta.name}/{file_upload_meta.file_name}"],
+                dest=f"{file_upload_meta.base_dir}/downloaded",
+            )
+            self._verify_downloaded_file_identical(
+                original_file_path=file_upload_meta.full_path,
+                downloaded_file_path=f"{file_upload_meta.base_dir}/downloaded/{file_upload_meta.file_name}",
+            )
+        finally:
+            await session.ComputeSession(name=test_name).destroy()
+            if vfolder_meta:
+                await self._cleanup_dummy_file(file_upload_meta)
```

Comment:
I think we could separate this part into a Wrapper and apply it.
Could you separate the process of receiving vfolder_meta and creating a session into a wrapper, and the download operation?
Please modify the template separation because it is also for easily tracking at which stage it failed. @seedspirit